pkg/operation/botanist/secrets.go

```diff
@@ -58,7 +58,15 @@ func (b *Botanist) GenerateAndSaveSecrets(ctx context.Context) error {
 		}
 	}
 
-	if b.Shoot.Info.DeletionTimestamp == nil {
+	// remove operation annotation
+	if err := b.Shoot.UpdateInfo(ctx, b.K8sGardenClient.Client(), false, func(shoot *gardencorev1beta1.Shoot) error {
+		delete(shoot.Annotations, v1beta1constants.GardenerOperation)
```

Comment:
Earlier, this annotation was only removed in case of `gardener.cloud/operation: rotate-kubeconfig-credential` or `gardener.cloud/operation: rotate-ssh-keypar`. Doing it generally here might bring unexpected side effects in the future, for example when we introduce a new operation.

Reply:
Yes, I "improved" the code by eliminating the duplication, not noticing that I am introducing different behavior (and potentially a bug). I will "unimprove" it now. Thanks a lot for noticing!

Reply:
Done